agent/internal/crypt/pgp_test.go

```diff
@@ -0,0 +1,170 @@
+YTBXLKXGILlIvAiTpcpL1lQpG1IjBlja/LWUY1V/PYkIL2ZqZZm7We//BWUn
+q7We9qlOHDsCkVv7ZFXcexMhslO/m1B72QPCuG7yi2+K4vvMDvx+8FgXoRpO
+iAdUIUVzEwQA2JjXnw+XB1mTPSj3MkKwxbM9NxILcjmlNxJMBAgvPNOM+cPM
+BS0fVYxXI9PZuT8pin2XFFQjFAN+B00yI19Ejt5/Zd5ruXmzhIqotIyujc6w
+BRHWMTxeTKOyTxAy2r+eUnwUGU6YppqeDdSDtKdimde5t3QKgOsSjetZCh5+
+pksD/3ZdJUJOXhewRwjBQ9yBcqWhxoB6+MklGopsxeeHM2+9u6dsYfOWm4ed
+sWL3C6wJNZLJVf8DzU2OBKp+YtH6jz4oppzxc3Jo/skE5y8ePPsVbQ9zt2FT
+OCaCPb7+73Y6U9kZn73g+dtWmAzCKgiYjcUUzsejWHeb9T98P7C7H8h9OHbC
+wHYEGAEIAAkFAmNL+qkCGwwAIQkQRou5QXdtqbcWIQSfcm14mlnpQiQYYk1G
+i7lBd22ptw2aCACFZP7wil0K7DhiDLnvEM0qoDNaKthINPGZ+XhBmjXti6i0
+94W84JtUYkFU8YnY6KfM9+8jjkD+U20VIfHzPiknKGMU4W4a7TRBKYpUe46/
+gacPY12Du9ygLDUEi+1FZmf0/fFifv6LsKP7PNnSMIxdSFG6kzfp42ktGtHl
+LddKexkr3xuZpxfs2ToCzxptHFklXuOQDLkqBAH7a8C62Uj/grtrh83tMFku
+Y5T0DDLlM3fDyia8eaYy9wYBLImZ2Zgy3it+UmUHc5OpaeMM4NYhMB5V08iw
+meR0tVQWeMpaue0VtGLPKHPGOknElgItP7cyVWJqEr6oM/x7uNrTpl3a
+=BqNZ
+-----END PGP PRIVATE KEY BLOCK-----
+`  // encrypted private key for unit test
+	pubKey1 = `-----BEGIN PGP PUBLIC KEY BLOCK-----
+Version: OpenPGP.js v4.10.10
+Comment: https://openpgpjs.org
+
+xsBNBGNL/YwBCADoeS0lcePet5BS+nCWrM8LLFR3tFXz/q/5hwaiqZTXR6So
+559uq0htGHsxP0529EZlbftHGDCZDfQ79N3uEmCSNQXpCPfSwPeAupRolA+C
+pXvbXrqC/jA3415ukNBWD7PecLcpza7n88lQpsSqsxwNAEAKt44Gzg7D8vzK
+wcdWJkO4C6GrDELHSN5nOKSlnmZsQVbuJ6IPdb5RIq4pBwnGQjjQM1bI9IuX
+h7iM9o4kxTxQHhbZ6gFXvXsBEgbGatI0jR44fU1kxxF86wxcL4quc8pnkBjd
+zN501BZjxscjZZBzYdDjowR8H/a+3iF97OEXjU0ytf39VXXx1BECXj6bABEB
+AAHNFHRlc3QgPHRlc3RAdGVzdC5jb20+wsCNBBABCAAgBQJjS/2MBgsJBwgD
+AgQVCAoCBBYCAQACGQECGwMCHgEAIQkQT8F/790UG6kWIQQloKf3rmNF5eEz
+rYRPwX/v3RQbqTzaB/wPRhkAXgdGZ4bdS5NHDpONHHd+Kgz+2CaPg+BkTzI9
+9i/rpCtQIOWbGDObUkP16EKr7Kuqz8uaefckB+h4wadNvzlwfl2maBN4WKoO
+OaOlMme6qF0FLucvHNb4Nk1Rpq4EkZcIpKPhkrbieMDtDzsd5QAFWaGPsuJM
+D2DX3rC2LUZIjweqiaDUGsiDeYwZ7ocvB+KcK0A+j05Za28jh560MPETgghX
+N2JDV/6ijCR1r5uXDyjivGbbjG36YWwJG/PhuOxiCeeMcGoeZt4LSbG8mFOT
+ww4agYo+4BC6VOwAvIe1qHqWqS+qkux8aSG5d8JifKiLLlUWqTFm3yKlooNt
+zsBNBGNL/YwBCACtLvLwlJWwzhk1XO+ZPUkaqGTLMpds3N3/uxxunrQxMhZd
+bSeCy4Ski6+gAvwimYjoWg5xZUAt4u/TECxY08f87DjvffWN7mdUywNYWZ/S
+KuMSIfbGsbEuv10CPSbknfzGcqtgkAu1yJQhRh2am/o0EZzqbZKPYypAYgmo
+IEmTxkhRVPBTT49Jjd92KB6Ysn/vXftWhR9fqD7OhbwR82eOfbDo5dh0h6bF
+7u6zonRr5Uj7TCEgrGllnM19OpdvuVzSSK9YTCrOrFvqRjl1v77MqEa99vA5
+CqXFosfnNEDhUrLAY39SVc7AACxEDGXnfLemQflhqdmpeVRpoPt0381zABEB
+AAHCwHYEGAEIAAkFAmNL/YwCGwwAIQkQT8F/790UG6kWIQQloKf3rmNF5eEz
+rYRPwX/v3RQbqTy9B/9oUJR40+rkhwMEPblBjvbtFXJ5GSIb2gk2nfvRfnmq
+aR7npayqvuEpNE9bbgWRY85yXDhjzDMo0omlE1yH3/geQ7Dzg3X+u0SDuOEV
+dL+yz5t9m1C4Ns+xKtun2BEZ2yv8KoEBY0ccLEDKMPyySyeW1z3uB7J/klW2
+D8lPOwOlj3yhj8dZZ8ItDCOZMYOlYwS/Jdho8ZAwza5poE4GYfRYQhYR6wyS
+OkAba6Xu8xCJwZ0I+Z8321OP6Ra07NfOnmeQeEcHL66+8EEHt+Wne8sMDNL/
+m9i3nF7Bw3jnCioAbVJMpKecLMKUdCg0Cw5OnG7+1WSC+lwb2OEgY5V6hHan
+=Zkso
+-----END PGP PUBLIC KEY BLOCK-----
+`
+)
+
+//const passphrase = []byte(`the passphrase of the private key`) // Passphrase of the privKey
+
+func TestDecryptSecrets(t *testing.T) {
```

Comment:
In my opinion it would be better to separate the cases into multiple tests.

Reply:
split the good case and fail case to 2 functions

Reply:
@joremysh I think you have to split every scenario into different test cases. Collecting good test cases in 1 function as you mentioned is not the best solution.

Reply:
There's only one case in each test so it looks good to me 👍